rethinkdb.d.ts

```diff
@@ -1676,7 +1677,7 @@ declare namespace rethinkdb {
     }
 
     namespace Merge {
-      type MergeParam <T> = RSelectionObjectResult<T> | ((item: RObject<T>) => (RObject<T> | T)) | T;
+	  type MergeParam <T> = RSelectionObjectResult<T> | ((item: RObject<T>) => (RObject<T> | T | object)) | T;
```

Comment:
When submitting a PR, make sure you follow the existing style on indentation. However, I'm not sure I follow how these help. `object` will mean type information is all lost, whereas you could just do `<object>` for the same effect (input and output of `object` allowed). Do you have examples of this usage and maybe can add a test case?

Reply:
[Here it is](https://gist.github.com/ChrisTalman/aed01e5657ce593973dd6d74bf136b63). I hope this helps.

Reply:
The TS compiler throws an error like this:
```
Argument of type '(subject: RObject<{ subjectId: RDatum<string>; }>) => { authorised: RBoolean<boolean>; }' is not assignable to parameter of type 'MergeParam<{ subjectId: RDatum<string>; }>'. Type '(subject: RObject<{ subjectId: RDatum<string>; }>) => { authorised: RBoolean<boolean>; }' is not assignable to type '(item: RObject<{ subjectId: RDatum<string>; }>) => { subjectId: RDatum<string>; } | RObje...'. Type '{ authorised: RBoolean<boolean>; }' is not assignable to type '{ subjectId: RDatum<string>; } | RObject<{ subjectId: RDatum<string>; }>'. Type '{ authorised: RBoolean<boolean>; }' is not assignable to type 'RObject<{ subjectId: RDatum<string>; }>'. Property 'do' is missing in type '{ authorised: RBoolean<boolean>; }'.
```

Reply:
The `interface` for `ObjectConstructor` in the official [lib.es2015.core.d.ts](https://github.com/Microsoft/TypeScript/blob/master/lib/lib.es2015.core.d.ts#L295) might be of some guidance here.

It seems that one to three sources are typed using intersections, but beyond that type information is lost, because it uses a rest parameter with the `any` type. Perhaps there's a way to intersect a dynamic set of types?